data/fichera-hex.mesh

```diff
@@ -0,0 +1,56 @@
+MFEM mesh v1.0
```

Comment:
I am worried we are getting too many files in the `data/` directory... Can we move the three new `fichera` meshes and maybe `mixed.mesh` to the [data repo](https://github.com/mfem/data) instead?

If we want to have a mesh with all 4 element types, maybe @koning can suggest a simplified version of the meshes he is interested in?